layers/vk_api_hash.h

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2023 Nintendo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <functional>
+
+#include <vulkan/vulkan.h>
+
+namespace std {
+
+#define APPEND_HASH(var) res = res * 31 + std::hash<std::decay<decltype(var)>::type>()(var)
```

Comment:
It would be nice to have a comment here, specifically about the purpose of the 31 constant